### PR TITLE
add a helper in allocator to prevent the usage of the `new` operator

### PR DESCRIPTION
**Pull request by BBasile**
This could work for `delete` too but actually there's a bug: https://issues.dlang.org/show_bug.cgi?id=18198.
Since `delete` is likely to be deprecated it's acceptable not to handle it here.

### CODE REVIEW

File: std/experimental/allocator/package.d [hunk 1]
```diff
@@ -3113,3 +3139,4 @@ unittest
     b = alloc2.allocate(101);
     assert(alloc2.deallocate(b));
 }
+
```
Comment (wilzbach) [inline std/experimental/allocator/package.d]: Unrelated?

**Comment by wilzbach** [comment]
(rebased to `upstream/master` to fix the CI failures)

File: std/experimental/allocator/package.d (hunk 1 above)
Comment (BBasile) [reply std/experimental/allocator/package.d]: Yes, things were initially at the end, until i realize that because of the `__EOF__` the test were not executed and the ddoc not rendered.

**Comment by wilzbach** [comment]
After talking with @andralex shortly about it, here's his feedback:

- it can be reduced to `@disable new (size_t);` (no implementation needed)
- Phobos isn't about including one-lines `mixin(disableNew);` is exactly as long and still requires the import
- `@disable new (size_t);` is part of the language, `disableNew` will require people to lock it up in the docs.
- this lacks good use cases

-> I will close this now as our PR queue is exploding. If you or someone else comes up with a better motivation, please resubmit (or ping us for reopening).
